packages/xo-web/src/common/xo/index.js

```diff
@@ -2836,8 +2836,17 @@ export const selfBindLicense = ({ id, plan, oldXoaId }) =>
 export const subscribeSelfLicenses = createSubscription(() => _call('xoa.licenses.getSelf'))
 
 // Support --------------------------------------------------------------------
-
-export const checkXoa = () => _call('xoa.check')
+let promise
+export const checkXoa = () => {
+  // Debounce for 2 minutes
+  if(promise === undefined){
+    promise = _call('xoa.check')
+    setTimeout(() => {
+      promise = undefined
+    }, 120000)
+  }
+  return promise
+}
```

Comment:
As per our discussion, the debounce should be on `xo-server`'s (or `xo-server-xoa`'s) side :slightly_smiling_face: 

Also, I believe, you should implement `debounce` as a generic decorator instead of a specific implementation.